format/nexus.py

```diff
@@ -111,7 +111,7 @@ def visitor(name, obj):
                 numpy.string_("NXsubentry"),
             ]:
                 if "definition" in obj:
-                    if obj["definition"][()] == numpy.string_("NXmx"):
+                    if numpy.string_(obj["definition"][()]) == numpy.string_("NXmx"):
```

Comment:
If we _know_ that these strings are returned as bytes wouldn't
```suggestion
                    if obj["definition"][()] == b"NXmx":
```
be more natural?

Reply:
alternatively:
```suggestion
                    if obj["definition"][()] in ("NXmx", b"NXmx"):
```
if they can be both. Which would lend itself to be wrapped in a helper function such as
```python
def _h5match(obj, string):
    return obj[()] in (string, string.encode("utf-8"))

(...)
    if _h5match(obj["definition"], "NXmx"):
    (...)
```